README.md

```diff
@@ -10,12 +10,328 @@ A core widget library for Dojo 2.
 
 For more background on Widgets for Dojo 2, there is a document describing the [widgeting system](https://github.com/dojo/meta/blob/master/documents/Widget-System.md).
 
+- [Usage](#usage)
+- [Features](#features)
+    - [Base Widget](#base-widget)
+    	- [Simple Widgets](#simple-widgets)
+    	- [d](#d)
+    	- [Widgets with Children](#widgets-with-children)
+    - [Extending Base Widget](#extending-base-widget)
+    - [Projector](#projector)
+    - [Dojo Widget Components](#dojo-widget-components)
+- [How do I contribute?](#how-do-i-contribute)
+    - [Installation](#installation)
+    - [Testing](#testing)
+- [Licensing information](#licensing-information)
+
+## Usage
+
+To use dojo-widgets in a project install the package along with the required peer dependencies.
+
+```shell
+npm install dojo-widgets
+
+# peer dependencies
+npm install dojo-has
+npm install dojo-shim
+npm install dojo-core
+npm install dojo-compose
+npm install dojo-store
+```
+
+To use `dojo-widgets` import the module in the project. For more details see [features](#features) below.
+
+```ts
+import createButton from 'dojo-widgets/components/createButton';
+```
+
+or use the [dojo cli](https://github.com/dojo/cli-create-app) to create a complete Dojo 2 skeleton application.
+
 ## Features
 
 Dojo 2 Widgets are based on a virtual DOM implementation named [Maquette](http://maquettejs.org/) as well as some foundational classes
 provided in [dojo-compose](https://github.com/dojo/compose).
 
-To see how the widgets are utilised, there is an [example application](https://github.com/kitsonk/app-example) available.  Currently, outside of the source distribution itself, there is not a lot of documentation, as the code base is under heavy development.
+The examples below are provided in TypeScript syntax. The package does work under JavaScript, but for clarity, the examples will only include one syntax.
```

Comment:
Maybe "The package works fine with plain JavaScript (ES5 or ES6+), but for simplicity the examples are provided with only one syntax at this time"?